pkg/oc/bootstrap/clusterup/kubeapiserver/config.go

```diff
@@ -10,15 +10,19 @@ import (
 	"github.com/openshift/origin/pkg/oc/bootstrap/docker/dockerhelper"
 	"github.com/openshift/origin/pkg/oc/bootstrap/docker/run"
 	"github.com/openshift/origin/pkg/oc/errors"
-	"github.com/openshift/origin/pkg/oc/util/tmputil"
+	"github.com/openshift/origin/pkg/oc/util/dir"
 )
 
+const ComponentDirectoryName = "oc-cluster-up-kube-apiserver"
+
 type KubeAPIServerStartConfig struct {
 	// MasterImage is the docker image for openshift start master
 	MasterImage string
 	ImageFormat string
 	DNSPort     int
 	PublicHost  string
+
+	HostDir string
```

Comment:
I don't think we need to pass this through.  Skip the entire task if it is specified. If it is not specified, generated into a tempdir and if they want `--write-config`, they can specify it and we'll copy it out.